lib/ansible/modules/network/eos/eos_l2_interface.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Ansible by Red Hat, inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+DOCUMENTATION = """
+---
+module: eos_l2_interface
+version_added: "2.5"
+author: "Ricardo Carrillo Cruz (@rcarrillocruz)"
+short_description: Manage L@ interfaces on Arista EOS network devices.
+description:
+  - This module provides declarative management of L2 interfaces
+    on Arist EOS network devices.
+notes:
+  - Tested against EOS 4.15
+options:
+  name:
+    description:
+      - Name of the interface
+    required: true
+    aliases: ['interface']
+  mode:
+    description:
+      - Mode in which interface needs to be configured.
+    choices: ['access','trunk']
+  access_vlan:
+    description:
+      - Configure given VLAN in access port.
+        If C(mode=access), used as the access VLAN ID.
+  native_vlan:
+    description:
+      - Native VLAN to be configured in trunk port.
+        If C(mode=trunk), used as the trunk native VLAN ID.
+  trunk_allowed_vlans:
```

Comment:
Please add 
`alias: ['trunk_vlans']`

Bit confusing that the primary name is `trunk_allowed_vlans` though you use the aliased `trunk_vlans` in the examples.